### PR TITLE
Add the ability to have an archive extract if the hash has changed

### PR DESCRIPTION
**Pull request by magnusboman**

### CODE REVIEW

File: salt/states/archive.py [hunk 1]
```diff
@@ -32,6 +32,35 @@ def __virtual__():
         else False
 
 
+def updateChecksum(fname, target, checksum):
+    lines = []
+    compare_string = '{0}:{1}'.format(target, checksum)
+    if os.path.exists(fname):
+        with open(fname, 'r') as f:
```
Comment (cachedout) [inline salt/states/archive.py]: Please change these to `salt.utils.fopen`

**Comment by cachedout** [comment]
Hi @magnusboman. Thanks for the PR. The approach here looks sound. There is one small style issue in-line in this code and it looks like a few tests are failing. Could you please take a look? Please let me know if you have an questions.

**Comment by cachedout** [comment]
@magnusboman Did you force push a change to your branch? This looks like it now contains many non-related commits.

**Comment by magnusboman** [comment]
@cachedout I don't know that I forced anything but yeah, I did something wrong. Should I just cancel this PR and create a new one?
I tried to find the "style issue" you mentioned but not sure where to look... Can you point me in the right direction

**Comment by cachedout** [comment]
@magnusboman Yes, let's start over with a new PR. The issue I mentioned was to have you please change the calls to Python's `open` to instead use `salt.utils.fopen`.

**Comment by magnusboman** [comment]
New PR submitted, hence closing this one

**Pull request closed without merging**